services/spar/src/Spar/API.hs

```diff
@@ -228,15 +228,27 @@ idpGetAll zusr = withDebugLog "idpGetAll" (const Nothing) $ do
   _idplProviders <- wrapMonadClientWithEnv $ Data.getIdPConfigsByTeam teamid
   pure IdPList {..}
 
-idpDelete :: Maybe UserId -> SAML.IdPId -> Spar NoContent
-idpDelete zusr idpid = withDebugLog "idpDelete" (const Nothing) $ do
+idpDelete :: Maybe UserId -> SAML.IdPId -> Maybe Bool -> Spar NoContent
+idpDelete zusr idpid purge = withDebugLog "idpDelete" (const Nothing) $ do
   idp <- SAML.getIdPConfig idpid
   _ <- authorizeIdP zusr idp
   let issuer = idp ^. SAML.idpMetadata . SAML.edIssuer
       team = idp ^. SAML.idpExtraInfo . wiTeam
-  -- fail if idp is not empty
+  -- if idp is not empty: fail or purge
   idpIsEmpty <- wrapMonadClient $ isNothing <$> Data.getSAMLAnyUserByIssuer issuer
-  unless idpIsEmpty $ throwSpar SparIdPHasBoundUsers
+  when (not idpIsEmpty) $ do
+    if fromMaybe False purge
+      then do
+        let go :: Spar ()
+            go = do
```

Comment:
can we do the same without all the nesting?  at least the purge we can do with viewpattern

Reply:
496b0c4c504996e36adf0e701f608089f2f056f1